tests/e2e/per_site_exporter_test.go

```diff
@@ -0,0 +1,357 @@
+			testHostname := strings.Split(strings.TrimPrefix(testServer.URL, "http://"), ":")[0]
+			testURL := testServer.URL + "?" + generateCacheBuster("per-site-metrics-test")
+
+			By("Making HTTP requests through the proxy")
+			for i := 0; i < 3; i++ {
+				resp, _, err := testhelpers.MakeProxyRequest(client, testURL+fmt.Sprintf("&req=%d", i))
+				Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Request %d should succeed", i))
+				resp.Body.Close()
+			}
+
+			time.Sleep(5 * time.Second)
+
+			Eventually(func() bool {
+				metricsContent, err := getPerSiteMetrics()
+				if err != nil {
+					return false
+				}
+				requestMetric, err := getPerSiteMetricsValue(metricsContent, "squid_site_requests_total", testHostname)
+				if err != nil {
+					return false
+				}
+				return requestMetric >= 3
+			}, timeout*2, interval).Should(BeTrue(), "Per-site request metrics should increase after proxy traffic")
+		})
+
+		It("should expose bandwidth metrics per site", func() {
+			testHostname := strings.Split(strings.TrimPrefix(testServer.URL, "http://"), ":")[0]
+			testURL := testServer.URL + "?" + generateCacheBuster("per-site-bandwidth-test")
+			resp, body, err := testhelpers.MakeProxyRequest(client, testURL)
+			Expect(err).NotTo(HaveOccurred(), "Request should succeed")
+			resp.Body.Close()
+			Expect(len(body)).To(BeNumerically(">", 0), "Should receive response body")
+
+			time.Sleep(5 * time.Second)
+
+			Eventually(func() bool {
+				metricsContent, err := getPerSiteMetrics()
+				if err != nil {
+					return false
+				}
+				bytesMetric, err := getPerSiteMetricsValue(metricsContent, "squid_site_bytes_total", testHostname)
+				if err != nil {
+					return false
+				}
+				return bytesMetric > 0
+			}, timeout*2, interval).Should(BeTrue(), "Per-site bandwidth metrics should be recorded")
+		})
+	})
+
+	Describe("Per-Site Metrics Format and Standards", func() {
+		It("should return per-site metrics in valid Prometheus format", func() {
+			metricsURL := fmt.Sprintf("https://%s.%s.svc.cluster.local:9302/metrics", serviceName, namespace)
+			c := newHTTPSClient(10 * time.Second)
+
+			resp, err := c.Get(metricsURL)
+			Expect(err).NotTo(HaveOccurred(), "Should get per-site metrics response")
+			defer resp.Body.Close()
+			Expect(resp.StatusCode).To(Equal(http.StatusOK), "Per-site metrics endpoint should return 200")
+
+			contentType := resp.Header.Get("Content-Type")
+			expected := []string{
+				"text/plain; version=0.0.4; charset=utf-8",
+				"text/plain; version=0.0.4; charset=utf-8; escaping=values",
+				"text/plain; version=0.0.4; charset=utf-8; escaping=underscores",
+			}
+			Expect(expected).To(ContainElement(contentType))
+
+			body, err := io.ReadAll(resp.Body)
+			Expect(err).NotTo(HaveOccurred(), "Should read per-site metrics body")
+			metricsContent := string(body)
+
+			lines := strings.Split(metricsContent, "\n")
+			for _, line := range lines {
```

Comment:
It looks like this one is mostly testing the prometheus go library. I'm not sure we want to do that.